dask/dataframe/io/parquet.py

```diff
@@ -693,78 +665,58 @@ def _read_pyarrow(fs, fs_token, paths, columns=None, filters=None,
-        non_empty_pieces = sorted(
-            non_empty_pieces, key=lambda piece: natural_sort_key(piece.path))
-
-    # Determine divisions
-    if len(index_names) == 1:
-
-        # Look up storage name of the single index column
-        divisions_names = [storage_name for storage_name, name
-                           in storage_name_mapping.items()
-                           if index_names[0] == name]
-
-        if divisions_names:
-            divisions_name = divisions_names[0]
-        else:
-            divisions_name = None
-    else:
-        divisions_name = None
-
-    divisions = _get_pyarrow_divisions(non_empty_pieces, divisions_name,
-                                       schema, infer_divisions)
+    pieces = sorted(dataset.pieces, key=lambda piece: natural_sort_key(piece.path))
 
-    # Build task
     dtypes = _get_pyarrow_dtypes(schema, categories)
     dtypes = {storage_name_mapping.get(k, k): v for k, v in dtypes.items()}
 
     meta = _meta_from_dtypes(all_columns, dtypes, index_names,
                              column_index_names)
     meta = clear_known_categories(meta, cols=categories)
 
-    if out_type == Series:
-        assert len(meta.columns) == 1
-        meta = meta[meta.columns[0]]
-
-    task_name = 'read-parquet-' + tokenize(fs_token, paths, all_columns)
-
-    if non_empty_pieces:
-        task_plan = {
-            (task_name, i): (_read_pyarrow_parquet_piece,
-                             fs,
-                             piece,
-                             column_names,
-                             index_names,
-                             out_type == Series,
-                             dataset.partitions,
-                             categories)
-            for i, piece in enumerate(non_empty_pieces)
-        }
+    if gather_statistics:
+        # Read from _metadata file
+        if dataset.metadata and dataset.metadata.num_row_groups == len(dataset.pieces):  # one rg per piece
+            row_groups = [
+                dataset.metadata.row_group(i)
+                for i in range(dataset.metadata.num_row_groups)
+            ]
+        # Read from each individual piece (possibly slow)
+        else:
+            row_groups = [
+                piece.get_metadata(lambda fn: pq.ParquetFile(fs.open(fn, mode='rb'))).row_group(0)
+                for piece in dataset.pieces
+            ]
+
+        stats = []
+        for row_group in row_groups:
+            s = {'num-rows': row_group.num_rows, 'columns': []}
+            for i, name in enumerate(schema.names):
+                column = row_group.column(i)
+                d = {'name': name}
+                if column.statistics:
+                    d.update({'min': column.statistics.min,
+                              'max': column.statistics.max,
+                              'null_count': column.statistics.null_count})
+                s['columns'].append(d)
+            stats.append(s)
```

Comment:
I think that this would add too much complexity.  If you have suggestions on how to do it well then those would be welcome, though probably in another issue.

Reply:
Coming back to this from before...
I think an implementation would amount to either a subclass of DataFrame which knows how to do len or, probably better, an optimisation specific to parquet when a len is requested - along the lines of what the previous column-selection was doing.

Reply:
I have no objection in principle to tracking more metadata, though I do think that it will be hard to do in general.

Would you mind if we defer this conversation to future work?

Reply:
Certainly, was just scrolling through the comments and noticed this. I think generally it would be very useful to Dask to pass on more information where possible, to avoid doing extra work.